Subsystems/nosSettingsSubsystem/Source/nosSettingsSubsystem.cpp

```diff
@@ -0,0 +1,313 @@
+		registeredModuleInfo.SettingsList = nos::Table<nos::sys::settings::editor::SettingsList>::From(*list);
+		nosEngine.SendCustomMessageToEditors(NOS_NAME_STATIC(NOS_SETTINGS_SUBSYSTEM_NAME), registeredModuleInfo.SettingsList);
+
+		return NOS_RESULT_SUCCESS;
+	}
+	nosResult UnregisterEditorSettings() {
+		nosModuleInfo module = {};
+		if (nosEngine.GetCallingModule(&module) != NOS_RESULT_SUCCESS)
+			return NOS_RESULT_FAILED;
+		std::unique_lock lock(GSettingsDataManager->RegisteredModulesMutex);
+		if (auto it = GSettingsDataManager->RegisteredModules.find(nos::Name(module.Id.Name).AsString()); it != GSettingsDataManager->RegisteredModules.end())
+		{
+			settingsFb::editor::TSettingsList list;
+			list.module_name = it->second.SettingsList->module_name()->str();
+			nosEngine.SendCustomMessageToEditors(NOS_NAME_STATIC(NOS_SETTINGS_SUBSYSTEM_NAME), nos::Buffer::From(list));
+			GSettingsDataManager->RegisteredModules.erase(it);
+			return NOS_RESULT_SUCCESS;
+		}
+		nosEngine.LogE("Module %s is not registered for editor settings, but called UnregisterEditorSettings().", nosEngine.GetString(module.Id.Name));
+		return NOS_RESULT_FAILED;
+	}
+	void OnEditorConnected(uint64_t editorId) {
+		for (auto& [moduleName, registeredModuleInfo] : GSettingsDataManager->RegisteredModules) {
+			nosEngine.SendCustomMessageToEditors(NOS_NAME_STATIC(NOS_SETTINGS_SUBSYSTEM_NAME), registeredModuleInfo.SettingsList);
+		}
+	}
+	void OnMessageFromEditor(uint64_t editorId, nosBuffer message) {
+		auto& regModules = GSettingsDataManager->RegisteredModules;
+		auto msg = flatbuffers::GetRoot<settingsFb::editor::SettingsUpdateFromEditor>(message.Data);
+		auto moduleName = msg->module_name()->c_str();
+		auto it = regModules.find(moduleName);
+		if (it == regModules.end()) {
+			nosEngine.LogE("Module %s is not registered for editor settings, but Editor sent a message.", moduleName);
+			return;
+		}
+
+		it->second.itemUpdateCallback(nos::Name(msg->item_name()->c_str()).ID, *nos::Buffer(msg->data()).GetInternal());
+	}
+
+	nosResult OnRequest(uint32_t minor, void** outSubsystemCtx)
+	{
+		if (auto it = GExportedSubsystemVersions.find(minor); it != GExportedSubsystemVersions.end())
+		{
+			*outSubsystemCtx = it->second.get();
+			return NOS_RESULT_SUCCESS;
+		}
+		nosSettingsSubsystem& subsystem = *(GExportedSubsystemVersions[minor] = std::make_unique<nosSettingsSubsystem>());
+		subsystem.ReadSettings = [](nosName typeName, nosBuffer* buffer) -> nosResult { return GSettingsDataManager->ReadSettings(typeName, buffer); };
+		subsystem.WriteSettings = [](nosName typeName, nosBuffer buffer, nosSettingsFileDirectory directory) -> nosResult { return GSettingsDataManager->WriteSettings(typeName, buffer, directory); };
+		subsystem.RegisterEditorSettings = [](const nosSettingsList* list, nosPfnSettingsItemUpdate itemUpdateCallback) { return GSettingsDataManager->RegisterEditorSettings(list, itemUpdateCallback); };
```

Comment:
`ItemUpdateCallback` should return a `nosResult`, and if it is successful, the work of writing settings to the file should be handled by the settings subsystem.